apps/bam.iobio/src/util.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ *  This program and the accompanying materials are made available under the terms of
+ *  the GNU Affero General Public License v3.0. You should have received a copy of the
+ *  GNU Affero General Public License along with this program.
+ *   If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ *  SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ *  TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ *  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ *  IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+export const iobioURL = 'https://s3.amazonaws.com/iobio/NA12878/NA12878.autsome.bam';
+
+export const BamDisplayNames = {
+	mapped_reads: 'Mapped Reads',
+	forward_strands: 'Forward Strands',
+	proper_pairs: 'Proper Pairs',
+	singletons: 'Singletons',
+	both_mates_mapped: 'Both Mates Mapped',
+	duplicates: 'Duplicates',
+	coverage_depth: 'Coverage Depth',
+	coverage_hist: 'Read Coverage Distribution',
+	frag_hist: 'Fragment Length',
+	length_hist: 'Read Length',
+	mapq_hist: 'Mapping Quality',
+	baseq_hist: 'Base Quality',
+};
+
+export const defaultBamContext = {
+	mapped_reads: true,
+	forward_strands: true,
+	proper_pairs: true,
+	singletons: true,
+	both_mates_mapped: true,
+	duplicates: true,
+	coverage_depth: true,
+	coverage_hist: true,
+	frag_hist: true,
+	length_hist: true,
+	mapq_hist: true,
+	baseq_hist: true,
+};
+
+export type BamContext = typeof defaultBamContext;
+
+export type BamKey = keyof typeof defaultBamContext;
+
+export const BamDataKeys = Object.keys(defaultBamContext) as Array<BamKey>;
+
+export const percentKeys = BamDataKeys.slice(0, 6);
+
+export const histogramKeys = BamDataKeys.slice(7);
```

Comment:
slice is maybe not the most expressive DX? 
for knowing at a glance which keys are included

Reply:
Array de-structuring may help with that?

Reply:
Maybe but what I liked about slice is it's so succinct
Also in this syntax you have an unused coverage_depth var
There's no winning 🤷 

```
const [mapped_reads,
	forward_strands,
	proper_pairs,
	singletons,
	both_mates_mapped,
	duplicates,
	coverage_depth,
	coverage_hist,
	frag_hist,
	length_hist,
	mapq_hist,
	baseq_hist] = BamDataKeys;

  export const percentKeys = [mapped_reads,
    forward_strands,
    proper_pairs,
    singletons,
    both_mates_mapped,
    duplicates,];

  export const histogramKeys = [coverage_hist,
    frag_hist,
    length_hist,
    mapq_hist,
    baseq_hist];
```

Reply:
Oh also, make the lists alphabetical. Makes editing them and finding the terms you want in the future much easier.

Reply:
This is the typed declaration I was dreaming of but couldn't execute. `satisfies` is a new one and solves exactly some of the problems I was unsure how to navigate.

Reply:
updated: [Improved Types + Outlier handling](https://github.com/overture-stack/iobio-components/pull/24/commits/2fb3929436379660d3f94c780c7e1bf029f2292e)